assets/js/pages/Users/UsersPage.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { listUsers, deleteUser } from '@lib/api/users';
+
+import { toast } from 'react-hot-toast';
+
+import Users from './Users';
+
+const successDeleteMessage = 'User deleted successfully';
+const errorLoadingMessage = 'An error occurred during loading users';
+const errorDeletingMessage = 'An error occurred during deleting user';
+
+function UsersPage() {
+  const [loading, setLoading] = useState(false);
+  const [userData, setUserData] = useState([]);
+  const [error, setError] = useState(null);
+
+  const navigate = useNavigate();
+
+  const fetchUsers = () => {
+    setLoading(true);
+    listUsers()
+      .then((response) => {
+        setUserData(response.data);
+      })
+      .catch((_error) => {
+        setError(errorLoadingMessage);
+        setUserData([]);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  };
+
+  const onDeleteUser = (userId) => {
+    deleteUser(userId)
+      .then(() => {
+        fetchUsers();
+        toast.success(successDeleteMessage);
+      })
+      .catch((_error) => {
+        setError(errorDeletingMessage);
+      });
+  };
+
+  useEffect(() => {
+    fetchUsers();
+  }, []);
+
+  useEffect(() => {
```

Comment:
Do we need this? 
Not better to just `toast.error` in the 2 places we have it.
You would remove the `error` state

Reply:
I suggested to have the error state for consistency in all the fetching sequences we do and to have it, i think it's better for consistency, having an error state could give us more flexibility even in the future